Cassini/app/src/main/java/org/x/cassini/MainActivity.java

```diff
@@ -2,13 +2,133 @@
 
 import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.ImageView;
 import android.widget.TextView;
 
+import org.w3c.dom.Text;
+
 public class MainActivity extends AppCompatActivity {
 
+    private String TAG = "MainPage";
+    private TextView allEntries, timelineView, tags, starred, stories, settings, newEntry;
+    private ImageView allEntriesIcon, timelineViewIcon, tagsIcon, starredIcon, storiesIcon, settingsIcon, goTo1, goTo2, goTo3, goTo4, goTo5, goTo6;
```

Comment:
TextView works but it seems more natural to use buttons to hold with both icon and text together. Otherwise it is needed to set onClickListener on both textview and imageview as users will be able to jump to the desired page by pressing either headers or icons